### ISSUE TITLE
Sensitive vertfile parser in sct_seg_gm

### ISSUE THREAD

**Issue by emilljungberg** [neuropoly/spinalcordtoolbox#929]
Hi, 

I just noticed a small thing in the sct_segment_graymatter when it parses the vertebrae levels file, mainly from trying it on the data from the segmentation challenge from ISMRM. In the file they provide they give vert levels as both numbers but also 2/3 if the slice is on the vert disc. It is understand if is not necessary to provide this information but the parser of the file crashes when this occurs. This is easy to trace back but something I ran into after this is that if the last line is not finished with a new line character that parser will again crash. 

Example:
```
Traceback (most recent call last):
  File "/Users/emil/scripts/sct_3.0_beta13/scripts/sct_segment_graymatter.py", line 694, in <module>
    gmsegfull.segment()
  File "/Users/emil/scripts/sct_3.0_beta13/scripts/sct_segment_graymatter.py", line 354, in segment
    self.segmentation_pipeline()
  File "/Users/emil/scripts/sct_3.0_beta13/scripts/sct_segment_graymatter.py", line 401, in segmentation_pipeline
    self.gm_seg = SupervisedSegmentationMethod(self.preprocessed.processed_target, self.level_to_use, self.model, gm_seg_param=self.seg_param)
  File "/Users/emil/scripts/sct_3.0_beta13/scripts/msct_multiatlas_seg.py", line 1168, in __init__
    self.segment()
  File "/Users/emil/scripts/sct_3.0_beta13/scripts/msct_multiatlas_seg.py", line 1183, in segment
    self.target_seg_methods = TargetSegmentationPairwise(self.model, target_image=self.im_target, levels_image=self.im_level, seg_param=self.param)
  File "/Users/emil/scripts/sct_3.0_beta13/scripts/msct_multiatlas_seg.py", line 822, in __init__
    self.process()
  File "/Users/emil/scripts/sct_3.0_beta13/scripts/msct_multiatlas_seg.py", line 835, in process
    list_levels = load_level(self.im_levels, type=self.model.param.use_levels, verbose=self.param.verbose)
  File "/Users/emil/scripts/sct_3.0_beta13/scripts/msct_gmseg_utils.py", line 509, in load_level
    level = int(level[:-1])
ValueError: invalid literal for int() with base 10: ''
```

The error is thrown from `msct_gmseg_utils.py` line 507
```
for line in lines[1:]:
   i_slice, level = line.split(',')
   level = int(level[:-1])                                # <---- This is where it crashes
   i_slice = int(i_slice)
   dic_level_by_i[i_slice] = level
```
I guess it occurs since `level` on the last line will be a string with 2 characters instead of 3 as the other lines. Thus, `level[:-1]` will give a whitespace character which can't be casted to an int (as seen above). Maybe a solution using regular expression could get around this by being less sensitive to the formatting of the file. The 2/3 thing is easy to instruct the user not to use but the end line character might be tricky to identify.

As always, thanks for awesome software guys!

**Comment by emilljungberg** [comment]
I guess this is fixed in the latest commit by you @SaraDupont ?

**Comment by SaraDupont** [comment]
Yes, it's fixed in dda20ed
